src/DynamoCore/Search/NodeSearchModel.cs

```diff
@@ -280,11 +293,20 @@ internal IEnumerable<NodeSearchElement> Search(string search, LuceneSearchUtilit
             return null;
         }
 
-        internal NodeSearchElement FindModelForNodeNameAndCategory(string nodeName, string nodeCategory, string parameters)
+        /// <summary>
+        /// Finds the node model that corresponds to the input nodeName, nodeCategory and parameters.
+        /// </summary>
+        /// <param name="nodeName"></param>
+        /// <param name="nodeCategory"></param>
+        /// <param name="parameters"></param>
+        /// <param name="ctk">Cancellation token to short circuit the operation.</param>
+        /// <returns></returns>
+        internal NodeSearchElement FindModelForNodeNameAndCategory(string nodeName, string nodeCategory, string parameters, CancellationToken ctk = default)
         {
             var result = Entries.Where(e => {
                 if (e.Name.Replace(" ", string.Empty).Equals(nodeName) && e.FullCategoryName.Equals(nodeCategory))
                 {
+                    ctk.ThrowIfCancellationRequested();
```

Comment:
Where is this caught? Why can't the `throw` be at the start of the `FindModel` function?

Reply:
I will put on at the beginning too